tests/e2e/vcp_to_csi_create_delete.go

```diff
@@ -502,6 +502,129 @@ var _ = ginkgo.Describe("[csi-vcp-mig] VCP to CSI migration create/delete tests"
+		15. Delete PVC2
+		16. verify PVC2, PV2 and the cns volume(vmdk) get deleted
+		17. Delete SC1 and SC2
+	*/
+	ginkgo.It("Statically provision VMDK used by a PV provisioned by VCP using CSI", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating VCP SC")
+		scParams := make(map[string]string)
+		scParams[vcpScParamDatastoreName] = GetAndExpectStringEnvVar(envSharedDatastoreName)
+		vcpScRetain, err := createVcpStorageClass(client, scParams, nil, v1.PersistentVolumeReclaimRetain, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		vcpScs = append(vcpScs, vcpScRetain)
+
+		ginkgo.By("Creating VCP PVC pvc1 before migration")
+		pvc1, err := createPVC(client, namespace, nil, "", vcpScRetain, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		vcpPvcsPreMig = append(vcpPvcsPreMig, pvc1)
+
+		ginkgo.By("Waiting for all claims created before migration to be in bound state")
+		vcpPvsPreMig, err = fpv.WaitForPVClaimBoundPhase(client, vcpPvcsPreMig, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Enabling CSIMigration and CSIMigrationvSphere feature gates on kube-controller-manager")
+		err = toggleCSIMigrationFeatureGatesOnKubeControllerManager(ctx, client, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		kcmMigEnabled = true
+
+		ginkgo.By("Waiting for migration related annotations on PV/PVCs created before migration")
+		waitForMigAnnotationsPvcPvLists(ctx, client, namespace, vcpPvcsPreMig, vcpPvsPreMig, true)
+
+		ginkgo.By("Verify CnsVSphereVolumeMigration crds and CNS volume metadata on pvc created before migration")
+		verifyCnsVolumeMetadataAndCnsVSphereVolumeMigrationCrdForPvcs(ctx, client, namespace, vcpPvcsPreMig)
+
+		vpath := getvSphereVolumePathFromClaim(ctx, client, namespace, pvc1.Name)
+		found, crd := getCnsVSphereVolumeMigrationCrd(ctx, vpath)
+		gomega.Expect(found).To(gomega.BeTrue())
+		fcdId := crd.Spec.VolumeID
+
+		ginkgo.By("Delete PVC1 and PV1")
+		err = client.CoreV1().PersistentVolumeClaims(namespace).Delete(ctx, pvc1.Name, *metav1.NewDeleteOptions(0))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		err = client.CoreV1().PersistentVolumes().Delete(ctx, vcpPvsPreMig[0].Name, *metav1.NewDeleteOptions(0))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		vcpPvcsPreMig = []*v1.PersistentVolumeClaim{}
+		vcpPvsPreMig = []*v1.PersistentVolume{}
+
+		ginkgo.By("Verify cnsvspherevolumemigrations crd are deleted")
+		err = waitForCnsVSphereVolumeMigrationCrdToBeDeleted(ctx, crd)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create CSI SC SC2")
+		csiSC, err := createStorageClass(client, nil, nil, v1.PersistentVolumeReclaimDelete, storagev1.VolumeBindingImmediate, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create PV2 statically via CSI with fcd leftover by PV1 using FCD ID noted earlier and SC2")
```

Comment:
We can make this as a separate function right?

Reply:
This part is specific for this testcase.
There is a generic method in util.go, `getPersistentVolumeSpec()` for normal usage